src/pybind/mgr/dashboard/requirements.txt

```diff
@@ -11,5 +11,5 @@ pyyaml
 natsort
 setuptools
 jsonpatch
-grpcio==1.46.5
-grpcio-tools==1.46.5
+grpcio
+grpcio-tools
```

Comment:
Thank you @pecastro . I was aware of the issue and I tried to fix it a while ago using a different PR: https://github.com/ceph/ceph/pull/55913.

The main problem I faced over there is that different distros supports different versions of grpc deps for eg: epel9 only [support grpcio 1.46.x](https://pkgs.org/search/?q=python3-grpcio) so far and other distros support some ubuntu has 1.51.x. These difference in version forces us to regenerate the protobuf file individually for every version of grpc or otherwise depending on the grpc version available that could break and completely dismantle the nvmeof functionality which is where we went with the current version pinning.

I think you might have already noticed that issue because I see you are also regenerating a new version of protobufs which could break the functionality in epel9 distros. Maybe rather than versioning the protobuf file within the repo, probably we should look into an alternate way to dynamically generate the file somehow during the controller init or while loading the dashboard module (or maybe even at the time of building the ceph). but will need to check that. @epuertat thoughts?

Reply:
Hi @nizamial09.

Thank you for the heads up.

My first objective here is to try to get a clean build ( and tests ) for a Fedora 40 box.
The former (build) has been achieved, now the latter (tests) is underway but in a limited fashion as my knowledge of Ceph and all of it's bits and parts is very shallow at best.

I've figured how to rebuild those protos because the error message was intuitive enough to point a possible direction.

I can definitely try to have a go at implementing one of your suggestions once the tests are passing and everyting is looking honky dory. I'll reach out to either you or @epuertat once I'm there.

`vstart.sh` still throws PyO3 modules errors when starting the manager which I don't think it should anymore and some of the tests are in a bit of a disarray let alone the other dragons I might find whilst I dig hence I'm not promising anything. :smiley:

Reply:
@pecastro on the PyO3 errors: I opened this PR https://github.com/ceph/ceph/pull/57926/ to fix it. It'll allow you to bootstrap the Ceph cluster without errors, but some follow-up actions might still trigger the PyO3 error.